driver-core/src/main/com/mongodb/internal/operation/ListCollectionsOperation.java

```diff
@@ -351,6 +375,9 @@ private BsonDocument getCommand() {
         }
         if (nameOnly) {
             command.append("nameOnly", BsonBoolean.TRUE);
+            if (authorizedCollections) {
+                command.append("authorizedCollections", BsonBoolean.TRUE);
+            }
```

Comment:
The documentation says: [_"When used without nameOnly: true, this option has no effect."_](https://docs.mongodb.com/manual/reference/command/listCollections/#:~:text=When%20used%20without%20nameOnly%3A%20true%2C%20this%20option%20has%20no%20effect.) That's why we send the option only if `nameOnly` is `true`.